src/frontend/komponenter/Fagsak/Brevmottaker/LeggTilEndreBrevmottakerModal.tsx

```diff
@@ -73,12 +80,13 @@ export const LeggTilEndreBrevmottakerModal: React.FC = () => {
     const [navnErPreutfylt, settNavnErPreutfylt] = React.useState(false);
 
     React.useEffect(() => {
-        const navnSkalVærePreutfylt = erMottakerBruker(skjema.felter.mottaker.verdi);
-        if (navnSkalVærePreutfylt !== navnErPreutfylt) {
-            skjema.felter.navn.validerOgSettFelt((navnSkalVærePreutfylt && bruker.navn) || '');
-            settNavnErPreutfylt(navnSkalVærePreutfylt);
-        }
-    }, [skjema.felter.mottaker.verdi]);
+        const { mottaker, navn, land } = skjema.felter;
+        const navnSkalVærePreutfylt = erMottakerBruker(mottaker.verdi);
+        navn.validerOgSettFelt(
+            navnSkalVærePreutfylt ? preutfyltNavnFixed(mottaker.verdi, land.verdi, bruker.navn) : ''
+        );
+        settNavnErPreutfylt(navnSkalVærePreutfylt);
+    }, [skjema.felter.mottaker.verdi, skjema.felter.land.verdi]);
```

Comment:
Her oppdaterer vi state baster på en state-endring. Slik jeg tolker dokumentasjonen er det er noe react-teamet ønsker at vi skal unngå: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Jeg synes forsåvidt dette ser greit ut og jeg vet ikke hvor mye av gammel kode vi skal begynne å rydde opp i i denne PRen. Du får ta en avgjørelse på hva du ønsker å gjøre her